projects/grpc-client-server/prober_server/main.go

```diff
@@ -22,26 +22,64 @@ type server struct {
 	pb.UnimplementedProberServer
 }
 
-func (s *server) DoProbes(ctx context.Context, in *pb.ProbeRequest) (*pb.ProbeReply, error) {
-	// TODO: support a number of repetitions and return average latency
-	start := time.Now()
-	_, _ = http.Get(in.GetEndpoint())	// TODO: add error handling here and check the response code
-	elapsed := time.Since(start)
-	elapsedMsecs := float32(elapsed / time.Millisecond)
-
-	return &pb.ProbeReply{LatencyMsecs: elapsedMsecs}, nil
+func (s *server) DoProbes(ctx context.Context, request *pb.ProbeRequest) (*pb.ProbeReply, error) {
+	// get the number of probes from the probe request
+	numberOfProbes := request.GetNumberOfProbes()
+	// fmt.Printf("DoProbes | numberOfProbes %v\n", numberOfProbes)
+
+	// initialise a total time
+	var totalTime time.Duration
+
+	// support a number of repetitions and return average latency
+	for i := 0; i < int(numberOfProbes); i++ {
+		startTime := time.Now()
+		// fmt.Printf("DoProbes | startTime %v\n", startTime)
+
+		// make the request to the endpoint
+		response, err := http.Get(request.GetEndpoint())
+
+		// if the request errors
+		if err != nil {
+			return nil, fmt.Errorf("error: DoProbes request failed: %v", err)
```

Comment:
This is an interesting question! Let's say you do 5 probes, and the first four succeed, and the last one fails. What's the most useful thing to return to the user in this case? What are they likely to want to know/do?

Reply:
i think a good thing would be to count the successes/fails
and then return the average of however many successes there were 
with extra information about how many failed including possibly the errors they gave